commit 5bae2265b733ec7f9cb850cd44531e1b345b2782
Author: Volker Krause <vkrause@kde.org>
Date:   Sat Jul 26 15:55:40 2025 +0200

    Fix downloading to temporary file
    
    Regression from c1ef9310b30b1fefa789756cdaafb12d6cbcb187, which removed
    actually opening the temporary file we are about to write to.
    
    (cherry picked from commit b4bc6015ba7d175f1ed135434fd92f1a60f2b556)

diff --git a/Quotient/jobs/downloadfilejob.cpp b/Quotient/jobs/downloadfilejob.cpp
index 718604b0..267c0485 100644
--- a/Quotient/jobs/downloadfilejob.cpp
+++ b/Quotient/jobs/downloadfilejob.cpp
@@ -158,7 +158,7 @@ BaseJob::Status DownloadFileJob::prepareResult()
     } else {
         if (d->encryptedFileMetadata.has_value()) {
             std::unique_ptr<QFile> tempTempFile = std::make_unique<QTemporaryFile>();
-            if (!tempTempFile->isWritable()) {
+            if (!tempTempFile->open(QFile::WriteOnly)) {
                 qCWarning(JOBS) << "Failed to open temporary file for decryption"
                                 << tempTempFile->errorString();
                 return { FileError, "Couldn't open temporary file for decryption"_L1 };
